File: src/main/java/org/iot/dht/repository/DHTProjection.java
package org.iot.dht.repository;

import org.iot.dht.data.DHT;
import org.springframework.data.rest.core.config.Projection;

@Projection(name = "dht", types = {DHT.class})
public interface DHTProjection {
    Long getTimestamp();
    Double getTemperature();
    Double getHumidity();
}

File: src/main/java/org/iot/dht/repository/DHTProjection.kt
package org.iot.dht.repository

import org.iot.dht.data.DHT
import org.springframework.data.rest.core.config.Projection

@Projection(name = "dht", types = [DHT::class])
interface DHTProjection {
    val timestamp: Long
    val temperature: Double
    val humidity: Double
}